openapi.yaml

```diff
@@ -441,6 +443,39 @@ paths:
           description: Internal server error
       security:
         - api_key: []
+  /services/{serviceId}/history:
+    get:
+      tags:
+        - service-history
+      summary: Retrieve service history
+      description: Retrieve all versions of a service
+      operationId: getServiceHistory
+      parameters:
+        - name: serviceId
+          in: path
+          description: ID of the service
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: Service history fetched successfully
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/Service'
```

Comment:
I think it needs to be "paginated"

Reply:
Correct, and by the way, I'll take the opportunity to add the `last_update` property in the `Service` model

Reply:
8ff4109

Reply:
if we use cosmos there already exists the out-of-the-box _\_ts_ field wich contains the last updated timestamp (but is at a second level of granularity) and [is already indexed by default](https://learn.microsoft.com/en-us/azure/cosmos-db/index-policy#includeexclude-strategy). So, theoretically, we can efficiently query by partitionKey (_serviceId_) sorted by updateDate (_\_ts_), but remains the offset "problem" of the pagination

Reply:
You're right, with CosmosDB we'll have only the `continuation` token concept to implement pagination...we need to discuss it later